hasher-matcher-actioner/hmalib/common/configs/tx_apis.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+DynamoDB backed SignalExchangeAPI objects.
+"""
+
+from dataclasses import dataclass
+import typing as t
+from enum import Enum
+
+from botocore.exceptions import ClientError
+from jinja2 import ModuleLoader
+from threatexchange.exchanges.signal_exchange_api import SignalExchangeAPI
+
+from hmalib.common.config import HMAConfig, create_config, update_config
+from hmalib.common.logging import get_logger
+from hmalib.common.mappings import full_class_name, import_class
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class ToggleableSignalExchangeAPIConfig(HMAConfig):
+    signal_exchange_api_class: str
+
+    # Allow soft disables
+    enabled: bool = True
+
+    @staticmethod
+    def get_name_from_type(signal_exchange_api: t.Type[SignalExchangeAPI]) -> str:
+        return f"signal_exchange_api:{full_class_name(signal_exchange_api)}"
```

Comment:
(ignorable): I went through all that trouble to guarantee you human readable names, and you throw it away! :P

Reply:
This is only for config storage. Not shown to humans.